src/code/z_message.c

```diff
@@ -178,22 +509,276 @@ void func_80148CBC(PlayState* play, UNK_PTR puParm2, u8 arg2) {
+            msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2] = '9';
+        }
+        Font_LoadCharNES(play, msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2],
+                         msgCtx->unk120C4 + (msgCtx->unk120C2 << 7));
+        Audio_PlaySfx(NA_SE_SY_RUPY_COUNT);
+    } else if (msgCtx->stickAdjY >= 30) {
+        msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2]++;
+        if (msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2] > '9') {
+            msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2] = '0';
+        }
+        Font_LoadCharNES(play, msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2],
+                         msgCtx->unk120C4 + (msgCtx->unk120C2 << 7));
+        Audio_PlaySfx(NA_SE_SY_RUPY_COUNT);
+    } else if ((msgCtx->stickAdjX >= 30) && !D_801CFD84) {
+        D_801CFD84 = true;
+        msgCtx->unk120C2++;
+        if (msgCtx->unk120C2 > 2) {
+            msgCtx->unk120C2 = 2;
+        } else {
+            Audio_PlaySfx(NA_SE_SY_CURSOR);
+        }
+    } else if ((msgCtx->stickAdjX <= -30) && !D_801CFD84) {
+        D_801CFD84 = true;
+        msgCtx->unk120C2--;
+        if (msgCtx->unk120C2 < 0) {
+            msgCtx->unk120C2 = 0;
+        } else {
+            Audio_PlaySfx(NA_SE_SY_CURSOR);
+        }
+    } else {
+        D_801CFD84 = false;
+    }
+
+    msgCtx->bankRupeesSelected = (msgCtx->decodedBuffer.schar[msgCtx->unk120C0] * 100) - ('0' * 100);
+    msgCtx->bankRupeesSelected += (msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + 1] * 10) - ('0' * 10);
+    msgCtx->bankRupeesSelected += (msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + 2]) - '0';
+}
+
+void func_80149048(PlayState* play) {
+    MessageContext* msgCtx = &play->msgCtx;
+
+    if (msgCtx->stickAdjY <= -30) {
+        msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2]--;
+        if (msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2] < 0x30) {
+            msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2] = 0x39;
+        }
+        Font_LoadCharNES(play, msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2],
+                         msgCtx->unk120C4 + (msgCtx->unk120C2 << 7));
+        Audio_PlaySfx(NA_SE_SY_RUPY_COUNT);
+    } else if (msgCtx->stickAdjY >= 30) {
+        msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2]++;
+        if (msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2] >= 0x3A) {
+            msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2] = 0x30;
+        }
+        Font_LoadCharNES(play, msgCtx->decodedBuffer.schar[msgCtx->unk120C0 + msgCtx->unk120C2],
+                         msgCtx->unk120C4 + (msgCtx->unk120C2 << 7));
+        Audio_PlaySfx(NA_SE_SY_RUPY_COUNT);
+    }
+
+    msgCtx->bankRupeesSelected = (msgCtx->decodedBuffer.schar[msgCtx->unk120C0] * 10) - 0x1E0U;
```

Comment:
You would recognize it better as
`msgCtx->bankRupeesSelected = (msgCtx->decodedBuffer.schar[msgCtx->unk120C0] - 0x30) * 10;`
Note i'll change it to this (it matches) and use `'0'` instead of `0x30`

Reply:
Ah, that makes much more sense. Nice catch